domain/info.go

```diff
@@ -180,20 +182,31 @@ func (is *InfoSyncer) RemoveMinStartTS() {
 }
 
 // ReportMinStartTS reports self server min start timestamp to ETCD.
-func (is *InfoSyncer) ReportMinStartTS() {
+func (is *InfoSyncer) ReportMinStartTS(store kv.Storage) {
 	if is.manager == nil {
 		// Server may not start in time.
 		return
 	}
 	pl := is.manager.ShowProcessList()
-	var minStartTS uint64 = math.MaxUint64
+
+	// Calculate the lower limit of the start timestamp to avoid extremely old transaction delaying GC.
+	currentVer, err := store.CurrentVersion()
+	if err != nil {
+		logutil.BgLogger().Error("update minStartTS failed", zap.Error(err))
+		return
+	}
+	now := time.Unix(0, oracle.ExtractPhysical(currentVer.Ver)*1e6)
+	startTSLowerLimit := variable.GoTimeToTS(now.Add(-time.Duration(kv.MaxTxnTimeUse) * time.Millisecond))
+
+	minStartTS := variable.GoTimeToTS(now)
 	for _, info := range pl {
-		if info.CurTxnStartTS < minStartTS {
+		if info.CurTxnStartTS > startTSLowerLimit && info.CurTxnStartTS < minStartTS {
```

Comment:
Is there any test that covers these changes?

Reply:
Sorry but I'm not familiar with that part of code. Just thinking it will be nice if this is covered by tests.

Reply:
test case added, PTAL @MyonKeminta .